biogeochem/EDCanopyStructureMod.F90

```diff
@@ -1759,28 +1690,29 @@ subroutine leaf_area_profile( currentSite )
                            currentPatch%canopy_area_profile(cl,ft,iv)
                    end if
 
-                   if(currentPatch%tlai_profile(cl,ft,iv)>nearzero )then
-                      currentPatch%layer_height_profile(cl,ft,iv) = currentPatch%layer_height_profile(cl,ft,iv) &
-                           /currentPatch%tlai_profile(cl,ft,iv)
-                   end if
-
                 enddo
 
              enddo
           enddo
 
           ! --------------------------------------------------------------------------
           ! Set the mask that identifies which PFT x can-layer combinations have
-          ! scattering elements in them.
+          ! scattering elements in them for radiation.
+          ! RGK: I'm not sure we need nrad ... I can't see a scenario where
+          !      canopy_area_profile for these layers is not >0 for layers in ncan ...
+          !      Leaving this for the time being.
           ! --------------------------------------------------------------------------
-
+          currentPatch%canopy_mask(:,:) = 0
           do cl = 1,currentPatch%NCL_p
              do ft = 1,numpft
-                do  iv = 1, currentPatch%nrad(cl,ft)
+                do_leaflayer: do  iv = 1, currentPatch%nrad(cl,ft)
                    if(currentPatch%canopy_area_profile(cl,ft,iv) > 0._r8)then
                       currentPatch%canopy_mask(cl,ft) = 1
+                      exit do_leaflayer
+                   else
+                      exit do_leaflayer
```

Comment:
@rgknox I see you have some comments on the need for the nrad loop. Does this mean that the duplication of `exit do_leaflayer` in both the `then` and `else` blocks is intentional? If so, please feel free to resolve this conversation.

Reply:
I think it's doing what its supposed to be doing, but its just coded up poorly. Maybe I forgot to finish it up.  Since the uppermost leaf layer is always present, if any of the layers have area, we only need to query the top index (ie iv==). I'll make a correction here. I'll remove the loop, just have it query leaf layer 1, and double check that the array is initialized to zero.